src/backend_services.jl

```diff
@@ -1,45 +1,72 @@
-function _backend_services_create_jwt(config::BackendServicesConfig)
+function _backend_services_create_jwt(config::BackendServicesConfig, token_endpoint::AbstractString)
+    # Random string that uniquely identifies the JWT
     jti = Random.randstring(150)
 
-    now = TimeZones.now(TimeZones.localzone())
-    expiration_time = now + Dates.Minute(4)
-    expiration_time_seconds_since_epoch_utc = integer_seconds_since_the_epoch_utc(expiration_time)
-
+    # Expiration time (integer) in seconds since "epoch"
+    # SHALL be no more than 5 minutes in the future
+    expiration_time = Dates.now(Dates.UTC) + Dates.Minute(4)
+    expiration_time_seconds_since_epoch_utc = round(Int, Dates.datetime2unix(expiration_time))
+    
     jwt_payload_claims_dict = Dict(
-        "iss" => config.iss,
-        "sub" => config.sub,
-        "aud" => config.token_endpoint,
+        "iss" => config.client_id,
+        "sub" => config.client_id,
+        "aud" => token_endpoint,
         "jti" => jti,
         "exp" => expiration_time_seconds_since_epoch_utc,
     )
-    jwt = JSONWebTokens.encode(config.private_key, jwt_payload_claims_dict)
+    jwt = JWTs.JWT(; payload = jwt_payload_claims_dict)
+
+    # Sign
+    if config.keyid === nothing
+        JWTs.sign!(jwt, config.key)
+    else
+        JWTs.sign!(jwt, config.key, config.keyid)
+    end
+    @assert JWTs.issigned(jwt)
+    @assert config.keyid === nothing || JWTs.kid(jwt) == config.keyid
 
-    return jwt
+    return string(jwt)
 end
 
+# Ref: https://www.hl7.org/fhir/smart-app-launch/backend-services.html
 """
     backend_services(config::BackendServicesConfig)
 """
 function backend_services(config::BackendServicesConfig)
-    jwt = _backend_services_create_jwt(config)
+    # Retrieve the server configuration
+    # Ref: https://www.hl7.org/fhir/smart-app-launch/backend-services.html#retrieve-well-knownsmart-configuration
+    _config_response = HTTP.request(
+        "GET",
+        joinpath(config.base_url, ".well-known/smart-configuration");
+        # In principle, it should be possible to omit the header
+        # (and servers may ignore it anyway)
+        # Ref: https://www.hl7.org/fhir/smart-app-launch/conformance.html#using-well-known
+        headers = ("Accept" => "application/json",),
+    )
+    config_response = JSON3.read(_config_response.body)
```

Comment:
```suggestion
    if config_response.status !== 200
        config_response = JSON3.read(_config_response.body)
```
Fall back to /metadata if status is not 200.